tests/performance/reindexing/reindexing_and_feeding.rb

```diff
@@ -0,0 +1,254 @@
+			  :data => "{ \"fields\": { \"text\": \"GNU's not UNIX\" } }"
+		  } => {
+			  'qps' => { :y_min =>  2650, :y_max =>  3250 },
+			  '95p' => { :y_min =>   1.4, :y_max =>   1.7 }
+		  },
+		  {
+			  :legend => 'GET small data 4 clients',
+			  :fbench => { :clients => 4, :use_post => false }
+		  } => {
+			  'qps' => { :y_min =>  5600, :y_max =>  6250 },
+			  '95p' => { :y_min =>   0.7, :y_max =>   0.9 }
+		  },
+		  {
+			  :legend => 'POST large data 128 clients',
+			  :fbench => { :clients => 128, :use_post => true},
+			  :data => "{ \"fields\": { \"text\": \"GNU#{"'s not UNIX" * (1 << 10) }\" } }"
+		  } => {
+			  'qps' => { :y_min =>  6300, :y_max =>  6700 },
+			  '95p' => { :y_min =>    24, :y_max =>    28 }
+		  },
+		  {
+			  :legend =>  'GET large data 128 clients',
+			  :fbench => { :clients => 128, :use_post => false }
+		  } => {
+			  'qps' => { :y_min => 29500, :y_max => 33000 },
+			  '95p' => { :y_min =>   4.9, :y_max =>   5.5 }
+		  }
+	  }
+	  @graphs = get_graphs
+	  @app = SearchApp.new.monitoring("vespa", 60).
+		  container(Container.new("combinedcontainer").
+			    jvmargs('-Xms16g -Xmx16g').
+			    search(Searching.new).
+			    docproc(DocumentProcessing.new).
+			    gateway(ContainerDocumentApi.new)).
+	  admin_metrics(Metrics.new).
+	  indexing("combinedcontainer").
+	  sd(selfdir + "doc.sd")
+  end
+
+  def test_reindexing_performance_and_impact
+    deploy_app(@app)
+    start
+
+    # First time is a dummy.
+    trigger_reindexing
+    wait_for_reindexing
+
+    qrserver = @vespa.container["combinedcontainer/0"]
+    document_count = 1 << 20
+
+    initial_file = dirs.tmpdir + "initial.json"
+    puts "Writing initial data to " + initial_file
+    qrserver.write_document_operations(:put,
+				       { :fields => { :label => 'initial', :count => 0, :text => "FAST#{" Search and Transfer" * (1 << 6)}" } },
+				       'id:test:doc::',
+				       document_count,
+				       initial_file)
+
+    refeed_file = dirs.tmpdir + "refeed.json"
+    puts "Writing refeed data to " + refeed_file
+    qrserver.write_document_operations(:put,
+				       { :fields => { :label => 'refeed', :count => 0, :text => "FAST#{" Search and Transfer" * (1 << 6)}" } },
+				       'id:test:doc::',
+				       document_count / 2,
+				       refeed_file)
+
+    updates_file = dirs.tmpdir + "updates.json"
+    puts "Writing updates to " + updates_file
+    qrserver.write_document_operations(:update,
+				       { :fields => { :count => { :increment => 1 } } },
+				       'id:test:doc::',
+				       document_count,
+				       updates_file)
+
+    # Warmup and feed corpus
+    puts "Feeding initial data"
+    feed_data({ :file => initial_file })
+
+    # Benchmark pure reindexing
+    puts "Reindexing corpus"
+    sleep 2
+    now_seconds = Time.now.to_i
+    assert_hitcount("indexed_at_seconds:%3C#{now_seconds}&nocache", document_count)	# All documents should be indexed before now_seconds
```

Comment:
Consider making `%3C` and `%3E` a constant, or wrap raw string in url/percent-encoder

Reply:
Will consider.